File: docs/guides/php/src/searchWithOptionalFilters.php
<?php

require __DIR__.'/../vendor/autoload.php';
use Algolia\AlgoliaSearch\Api\SearchClient;
use Algolia\AlgoliaSearch\Model\Search\OptionalFilters;
use Algolia\AlgoliaSearch\Model\Search\SearchParamsObject;

$labels = []; // A list of labels
$reduceLabelsToFilters = function (array $labels): OptionalFilters {
    // Implement your logic here
    return new OptionalFilters();
};

$client = SearchClient::create('ALGOLIA_APPLICATION_ID', 'ALGOLIA_API_KEY');

$optionalFilters = $reduceLabelsToFilters($labels);
$searchParams = (new SearchParamsObject())
    ->setQuery('<YOUR_SEARCH_QUERY>')
    ->setOptionalFilters($optionalFilters)
;

$client->searchSingleIndex(
    '<YOUR_INDEX_NAME>',
    $searchParams,
);

File: docs/guides/ruby/searchWithOptionalFilters.rb
require "algolia"

def reduce_labels_to_filters(_labels)
  # Implement your logic here
  []
end

# A list of labels
labels = []

client = Algolia::SearchClient.create("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")

optional_filters = reduce_labels_to_filters(labels)
search_params = Algolia::Search::SearchParamsObject.new(
  query: "<YOUR_SEARCH_QUERY>",
  optionalFilters: optional_filters
)

client.search_single_index("<YOUR_INDEX_NAME>", search_params)
